WordPress/src/main/java/org/wordpress/android/ui/ActivityLauncher.java

```diff
@@ -205,6 +205,8 @@ public static void browsePostOrPage(Context context, SiteModel site, PostModel p
 
         // always add the preview parameter to avoid bumping stats when viewing posts
         String url = UrlUtils.appendUrlParameter(post.getLink(), "preview", "true");
+        // Custom domains are not properly authenticated due to a server side(?) issue, so this gets around that
+        url = url.replace(site.getUrl(), site.getUnmappedUrl());
```

Comment:
This was a stupid mistake by me. This code should only run for .com sites and the `null` case should be handled. Fix incoming!